jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -127,12 +130,40 @@ public class DeploymentScanner extends ContainerLifeCycle implements Scanner.Bul
     // old attributes prefix, now stripped.
     private static final String ATTRIBUTE_PREFIX = "jetty.deploy.attribute.";
 
+    private static final Pattern EE_ENVIRONMENT_NAME_PATTERN = Pattern.compile("ee(\\d+)");
+
+    /**
+     * A comparator that ranks names matching EE_ENVIRONMENT_NAME_PATTERN higher than other names,
+     * EE names are compared by EE number, otherwise simple name comparison is used.
+     */
+    protected static final Comparator<String> ENVIRONMENT_COMPARATOR = (e1, e2) ->
+    {
```

Comment:
This comparator will generate a lot of matchers, so perhaps take the returns where you can:
```suggestion
    {
        if (Objects.equals(e1, e2))
            return 0;
```

Reply:
Better yet, why not just create an Index<Integer> that is case insensitive and maps all the known environment names to an integer.    

```java
    Index<Integer> environmentOrdinals = new Index.Builder<Integer>()
        .caseSensitive(false)
        .with("static", 1)
        .with("core", 2)
        .with("ee9", 9)
        .with("ee10", 10)
        .with("ee11", 11)
        .build();
```

Reply:
In a conversation with @sbordet and @lorban this will be fixed in a followup PR.

I have an approach that will push the weight of the tracked environments (the ones allowed to be deployed to) via a change to the `DeploymentScanner.environmentConfig(String name, int weight)`, putting the weights into the XMLs instead.

That way the weights are not hardcoded, and can be adjusted by any user, and even add new environments (with their own weights) without the need to modify code.